src/Laravel/Eloquent/State/LinksHandler.php

```diff
@@ -37,22 +40,72 @@ public function handleLinks(Builder $builder, array $uriVariables, array $contex
                 $identifier = $uriVariables[$uriVariable];
 
                 if ($to = $link->getToProperty()) {
-                    $builder = $builder->where($builder->getModel()->getTable().'.'.$builder->getModel()->{$to}()->getForeignKeyName(), $identifier);
+                    $builder = $builder->where($builder->getModel()->{$to}()->getQualifiedForeignKeyName(), $identifier);
 
                     continue;
                 }
 
                 if ($from = $link->getFromProperty()) {
                     $relation = $this->application->make($link->getFromClass());
-                    $builder = $builder->getModel()->where($builder->getModel()->getTable().'.'.$relation->{$from}()->getForeignKeyName(), $identifier);
+                    $builder = $builder->getModel()->where($relation->{$from}()->getQualifiedForeignKeyName(), $identifier);
 
                     continue;
                 }
 
-                $builder->where($builder->getModel()->getTable().'.'.$link->getIdentifiers()[0], $identifier);
+                $builder->where($builder->getModel()->qualifyColumn($link->getIdentifiers()[0]), $identifier);
             }
 
             return $builder;
+        } elseif ($operation instanceof GraphQlOperation) {
+            if (!($linkClass = $context['linkClass'] ?? false)) {
+                return $builder;
+            }
+
+            $newLink = null;
+            $linkProperty = $context['linkProperty'] ?? null;
+
+            try {
+                $resourceMetadataCollection =  $this->application->make(ResourceMetadataCollectionFactoryInterface::class)->create($linkClass);
```

Comment:
can we use injection for this instead of calling the container? The container is useful to load Eloquent models but I'd like to avoid using it for something else (I don't know any other ways to load eloquent models, maybe you do?).

Reply:
Hm yeah you are right, i did it this way since we do the same thing above for HTTP operations. But you are right usually dependency injection is what's done in the codebase here, i will update the code.